packages/cli/src/lib/packager/index.ts

```diff
@@ -85,7 +90,11 @@ export async function createDistWorkspace(
 
   if (options.buildDependencies) {
     const scopeArgs = targets.flatMap(target => ['--scope', target.name]);
-    await run('yarn', ['lerna', 'run', ...scopeArgs, 'build'], {
+    const extraArgs = options.parallel
+      ? ['--parallel', options.parallel.toString()]
+      : [];
+
+    await run('yarn', ['lerna', 'run', ...scopeArgs, 'build', ...extraArgs], {
```

Comment:
This is just executing yarn `build` script of each package, so we can't really assume that the `--parallel` flag will work. I think it'd be better to add the `--parallel` flag to `backend:build`, and then just call the appropriate `build` scripts manually before `backend:build-image`.

Reply:
Yeah, I actually realized this isn't working as expected, which might be why it's still failing periodically. Was struggling a bit to get the parallel flag forwarded properly.

It really only needs to be applied to the app:build I think and not backend. Since the workspace uses temporary directories I don't think your suggestion of running the frontend build before build-image will work since the target dist will be different and the dockerfile won't pick it up. Any other ideas?

Reply:
To clarify, I this might be just to get the app to be served by the backend. If I remove the app plugin from the app plugin from the backend this might go away. But I would really like this to also work.

Reply:
The `dist` folder of the monorepo are what ends up being packed into the image, so it should work just fine as long as you run `build-image` without the `--build` flag.

Maybe it makes sense to make this switch with an `env` var instead then? Since that's much easier to carry on to the sub-commands.

Reply:
> The `dist` folder of the monorepo are what ends up being packed into the image, so it should work just fine as long as you run `build-image` without the `--build` flag.

@Rugvip - Sorry, I'm not completely following this. I realize the `--build` flag skips building the dependencies, but I'm not sure on the alternative here since there's a lot of logic going on in the createWorkspace which I'm not exactly sure all of what it does. It feels like if build manually first, then I need to replicate a bunch of things manually along with creating a custom dockerfile?

> Maybe it makes sense to make this switch with an `env` var instead then? Since that's much easier to carry on to the sub-commands.

This approach seems much more straightforward I think. I would basically, make it so the `app:build` also reads the parallel option from an env variable and/or the command line. Then the `backend:build-image` could set that environment variable before invoking the lerna commands. Is that what you had in mind?

Reply:
> > The `dist` folder of the monorepo are what ends up being packed into the image, so it should work just fine as long as you run `build-image` without the `--build` flag.
> 
> @Rugvip - Sorry, I'm not completely following this. I realize the `--build` flag skips building the dependencies, but I'm not sure on the alternative here since there's a lot of logic going on in the `createDistWorkspace` which I'm not exactly sure all of what it does. It feels like if I build manually first, then I need to replicate a bunch of things manually along with creating a custom dockerfile?

Think of it as `createDistWorkspace` takes all the `dist` folders in the repo and packages them into a single unit that can be deployed. Basically you can first `yarn build` (with `--parallel`) all the dependencies of the backend and the backend itself, and then call call `yarn backstage-cli build-workspace` to package it all.

> > Maybe it makes sense to make this switch with an `env` var instead then? Since that's much easier to carry on to the sub-commands.
> 
> This approach seems much more straightforward I think. I would basically, make it so the `app:build` also reads the parallel option from an env variable and/or the command line. Then the `backend:build-image` could set that environment variable before invoking the lerna commands. Is that what you had in mind?

You wouldn't even need to set it, the environment is (should be) forwarded when invoking sub commands.

Reply:
Thanks! I was not aware of `build-workspace` as an option. I'll probably implement the environment variable approach, but also try out the other suggested approach. I "think" between the 2 that should hopefully 🤞 should address the problem.

Reply:
For good measure I also added code to use the `lerna --concurrent [value]` global option when parallel is set to an integer.